filters/openpolicyagent/opaauthorizerequest/benchmark_test.go

```diff
@@ -232,50 +346,72 @@ func BenchmarkJwtValidation(b *testing.B) {
+		OpaControlPlaneUrl:  opaControlPlane.URL(),
+		DecisionConsumerUrl: "",
+		DecisionPath:        testDecisionPath,
+		BundleNames:         []string{testBundleName},
+		DecisionLogging:     false,
+		ContextExtensions:   "",
+		JwtCache:            true,
+	}
+
 	f, err := createOpaFilter(filterOpts)
+
 	require.NoError(b, err)
 
 	reqUrl, err := url.Parse("http://opa-authorized.test/somepath")
 	require.NoError(b, err)
 
-	claims := jwt.MapClaims{
-		"iss":   "https://some.identity.acme.com",
-		"sub":   "5974934733",
-		"aud":   "nqz3xhorr5",
-		"iat":   time.Now().Add(-time.Minute).UTC().Unix(),
-		"exp":   time.Now().Add(tokenExp).UTC().Unix(),
-		"email": "someone@example.org",
-	}
+	tokenCounts := []int{1, 5, 6, 10, 100}
+	for _, tokenCount := range tokenCounts {
+		tokens := make([]string, tokenCount)
+		for i := 0; i < tokenCount; i++ {
+			claims := jwt.MapClaims{
+				"iss":   "https://some.identity.acme.com",
+				"sub":   "5974934733",
+				"aud":   "nqz3xhorr5",
+				"iat":   time.Now().Add(-time.Minute).UTC().Unix(),
+				"exp":   time.Now().Add(tokenExp).UTC().Unix(),
+				"email": "someone@example.org",
+				"jti":   fmt.Sprintf("token-%d", i),
+			}
 
-	token := jwt.NewWithClaims(jwt.SigningMethodRS256, claims)
+			token := jwt.NewWithClaims(jwt.SigningMethodRS512, claims)
 
-	key, err := jwt.ParseRSAPrivateKeyFromPEM(privateKey)
-	require.NoError(b, err, "Failed to parse RSA PEM")
+			key, err := jwt.ParseRSAPrivateKeyFromPEM(privateKey)
+			require.NoError(b, err, "Failed to parse RSA PEM")
 
-	// Sign and get the complete encoded token as a string using the secret
-	signedToken, err := token.SignedString(key)
-	require.NoError(b, err, "Failed to sign token")
+			// Sign and get the complete encoded token as a string using the secret
+			signedToken, err := token.SignedString(key)
+			require.NoError(b, err, "Failed to sign token")
+			tokens[i] = signedToken
+		}
 
-	ctx := &filtertest.Context{
-		FStateBag: map[string]interface{}{},
-		FResponse: &http.Response{},
-		FRequest: &http.Request{
-			Header: map[string][]string{
-				"Authorization": {fmt.Sprintf("Bearer %s", signedToken)},
-			},
-			URL: reqUrl,
-		},
-		FMetrics: &metricstest.MockMetrics{},
-	}
+		b.Run(fmt.Sprintf("tokens_count_%d", tokenCount), func(b *testing.B) {
 
-	b.ResetTimer()
-	b.RunParallel(func(pb *testing.PB) {
-		for pb.Next() {
-			f.Request(ctx)
-			assert.False(b, ctx.FServed)
-		}
-	})
+			b.ResetTimer()
+			b.RunParallel(func(pb *testing.PB) {
+				for pb.Next() {
+					rnd := rand.New(rand.NewSource(time.Now().UnixNano() + int64(rand.Int())))
+					token := tokens[rnd.Intn(tokenCount)]
+					ctx := &filtertest.Context{
+						FStateBag: map[string]interface{}{},
+						FResponse: &http.Response{},
+						FRequest: &http.Request{
+							Header: map[string][]string{
+								"Authorization": {fmt.Sprintf("Bearer %s", token)},
+							},
+							URL: reqUrl,
+						},
+						FMetrics: &metricstest.MockMetrics{},
+					}
```

Comment:
I guess we could prepare an array of test contexts instead of tokens 
so that benchmark body only calls `f.Request(ctxs[rand.Intn(tokenCount)])`